collections-csharp/HashtableInCSharp/Tests/HashtableInCSharpTest.cs

```diff
@@ -5,71 +5,120 @@ namespace Tests
+
+            Assert.IsType<Hashtable>(userHashtable);
+            Assert.NotEmpty(userHashtable);
+            Assert.Equal(5, userHashtable.Count);
         }
 
         [Fact]
-        public void GivenAnEmptyHashtable_WhenRetrievingFromHashtable_ThenReturnsAUser()
+        public void GivenHashtable_WhenAddingSampleDataToHashtable_ThenReturnsHashTable()
         {
-            var user = _hashtableDemo.RetrieveAnElementFromHashTable(_users);
+            List<User> userList = new()
+            {
+                new User() { Id = 6, FirstName = "Judit", LastName = "Peter" },
+                new User() { Id = 7, FirstName = "Steve", LastName = "Billing" },
+                new User() { Id = 8, FirstName = "Goddy", LastName = "Opara" },
+            };
+
+            var populatedUserHashtable = HashtableDemo.AddSampleDataToHashTable(_userHashtable, userList);
+
+            Assert.IsType<Hashtable>(populatedUserHashtable);
+            Assert.NotEmpty(populatedUserHashtable);
+            Assert.Equal(8, populatedUserHashtable.Count);
+        }
+
+        [Fact]
+        public void GivenEmptyHashtable_WhenRetrievingElementFromHashtable_ThenReturnsUser()
+        {
+            var user = HashtableDemo.RetrieveElementFromHashTable(_userHashtable, 3);
             var userFirstName = user.FirstName;
             var userLastName = user.LastName;
 
-            Assert.Equal("Goddy", userFirstName);
-            Assert.Equal("Opara", userLastName);
+            Assert.Equal("Sam", userFirstName);
+            Assert.Equal("Manalt", userLastName);
         }
 
         [Fact]
-        public void GivenAnEmptyHashtable_WhenUpdatingElementInHashtable_ThenReturnsUpdatedHashTable()
+        public void GivenEmptyHashtable_WhenRetrievingAllElementFromHashtable_ThenReturnsAllUsers()
         {
-            var newUsers = _hashtableDemo.UpdateAnElementInHashTable(_users);
-            User user = (User)newUsers[1];
+            var users = HashtableDemo.RetrieveAllElementsFromHashTable(_userHashtable);
+
+            Assert.IsType<List<User>>(users);
+            Assert.NotEmpty(users);
+            Assert.Equal(5, users.Count);
+        }
+
+        [Fact]
+        public void GivenEmptyHashtable_WhenUpdatingElementInHashtable_ThenReturnsUpdatedHashTable()
+        {
+            var newUsers = HashtableDemo.UpdateElementInHashTable(_userHashtable, 3);
+            User user = (User)newUsers[3];
             var userFirstName = user.FirstName;
             var userLastName = user.LastName;
 
+            Assert.Equal(5, newUsers.Count);
             Assert.Equal("Henry", userFirstName);
             Assert.Equal("Stafford", userLastName);
         }
 
         [Fact]
-        public void GivenAnEmptyHashtable_WhenRemovingAnElementFromHashtable_ThenReturnsUpdatedHashTable()
+        public void GivenEmptyHashtable_WhenRemovingElementFromHashtable_ThenReturnsUpdatedHashTable()
```

Comment:
In this test, since you are verifying a removal of a particular element by key, there should be an assertion that the key exists before the removal, and then an assertion that the key no longer exists after the removal. Also, a size assertion is nice, but you should store the current size before the removal and you can assert that the new size is one less than the old size.